lib/cpp/src/thrift/server/TNonblockingServer.cpp

```diff
@@ -1524,7 +1532,7 @@ void TNonblockingIOThread::breakLoop(bool error) {
   }
 
   // sets a flag so that the loop exits on the next event
-  event_base_loopbreak(eventBase_);
+  event_base_loopexit(eventBase_, 0);
```

Comment:
Your commends correspond to the discussion in the Jira entry for this item from @nsuke.  Is that something you would have time to do?